src/main/scala/commonist/ui/ImageUI.scala

```diff
@@ -155,20 +155,35 @@ final class ImageUI(file:File, icon:Option[Icon], thumbnailMaxSize:Int, programH
 
 	// BETTER move unparsers and parsers together
 
-	private val	exif		= EXIF extract file
-	private val exifDate	= exif.date		cata ("", _ format "yyyy-MM-dd HH:mm:ss")
-	private val exifGPS		= exif.gps		cata ("", it => it.latitude.toString + "," + it.longitude.toString)
-	private val exifHeading	= exif.heading	cata ("", _.toString)
-	private val exifDesc	= exif.description getOrElse ""
-	private val fixedName	= Filename fix file.getName
-
-	uploadEditor		setSelected	false
-	nameEditor			setText		fixedName
-	descriptionEditor	setText		exifDesc
-	dateEditor			setText		exifDate
-	coordinatesEditor	setText		exifGPS
-	headingEditor		setText		exifHeading
-	categoriesEditor	setText		""
+	private def getExif():ImageData = {
+	  val	exif = EXIF extract file
+	  return new ImageData(
```

Comment:
`return` is not needed here, and ImageData is a case class so `new` should be left out, too.